tests/mail_test.sh

```diff
@@ -140,6 +140,18 @@ function test_mail_parser()
   assert_equals_helper 'Invalid option passed' "$LINENO" "$ret" 22
 
   # valid options
+  parse_mail_options some -- extra -1 args HEAD^
```

Comment:
Why `some` is valid here? I think it should fail since `some` is not a valid option.

Reply:
`some` does not start with a dash, so `kw_parse` assumes it is a non-option argument, like a file name, and sends it to the back of the arguments list, after the `--`. This is testing this precise behavior, if something like `--some` is passed instead, then we should see an error. I think there are tests for this already.